geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,411 @@
+            curves = [curves]
+
+        self._load_attributes()
+        for curve in curves:
+            curve = self._verify_object(curve, "Curve")
+            self.add_children(curve)
+            if self._interpretation_curves is None:
+                self._interpretation_curves = [curve]
+            elif curve not in self._interpretation_curves:
+                self._interpretation_curves.append(curve)
+
+        self._update_to_metadata("Interpretation curves", self._interpretation_curves)
+
+    def add_interpretation_section(
+        self, sections: InterpretationSectionParams | list[InterpretationSectionParams]
+    ):
+        """
+        Add an interpretation section.
+
+        :param sections: The interpretation section to add.
+        """
+        if not isinstance(sections, list):
+            sections = [sections]
+
+        self._load_attributes()
+        for section in sections:
+            section = InterpretationSectionParams.verify(section)
+            if self._interpretation_sections is None:
+                self._interpretation_sections = [section]
+            elif section.in_list(self._interpretation_sections) is None:
+                self._interpretation_sections.append(section)
+
+        self._update_to_metadata(
+            "Interpretation sections", self._interpretation_sections
+        )
+
+    @property
+    def can_add_group(self) -> bool:
+        """
+        Get the can add group attribute.
+
+        :return: The 'can add group' attribute.
+        """
+        return self._can_add_group
+
+    @can_add_group.setter
+    def can_add_group(self, value: bool):
+        """
+        Set the can add group attribute.
+
+        :param value: The can add group attribute.
+        """
+        self._can_add_group = bool(value)
+
+        if self.on_file:
+            self.workspace.update_attribute(self, "can_add_group")
+
+    @property
+    def interpretation_curves(self) -> list[Curve] | None:
+        """
+        Get the interpretation curves.
+
+        :return: The interpretation curves.
+        """
+
+        if self._interpretation_curves is None:
+            self.interpretation_curves = self._load_from_metadata(
+                "Interpretation curves"
+            )
+
+        return self._interpretation_curves
+
+    @interpretation_curves.setter
+    def interpretation_curves(self, curves: list[Curve] | None):
+        """
+        Set the interpretation curves.
+
+        :param curves: The interpretation curves.
+        """
+        if curves is None:
+            self._interpretation_curves = None
+        else:
+            if not isinstance(curves, list):
+                raise TypeError("Interpretation curves must be a list of Curve.")
+
+            self._interpretation_curves = [
+                self._verify_object(curve, "Curve") for curve in curves
+            ]
+
+            self.add_children(self._interpretation_curves)  # type: ignore
```

Comment:
Doesn't seem to do the trick right now
![image](https://github.com/user-attachments/assets/69c3d7c5-5ff0-4f0f-a6a5-e0f6d8496601)